### PR TITLE
doc blocks + fix deprecation utf8_encoding

### PR DESCRIPTION
**Pull request by ildyria**

### CODE REVIEW

File: app/Models/BaseAlbumImpl.php [hunk 1]
```diff
@@ -100,10 +100,14 @@
  * @property bool                       $grants_full_photo_access
  * @property bool                       $grants_download
  * @property Collection                 $shared_with
+ * @property int|null                   $shared_with_count
  * @property string|null                $password
  * @property bool                       $is_password_required
  * @property PhotoSortingCriterion|null $sorting
  * @property AlbumProtectionPolicy      $policy
+ * @property int                        $is_share_button_visible  // NOT USED
```
Comment (qwerty287) [inline app/Models/BaseAlbumImpl.php:108]: Why do we have this?

Comment (ildyria) [reply app/Models/BaseAlbumImpl.php:108]: I just know that it should be removed as we are no longer using it in our settings and we just use a global setting everywhere.
The update of doc block is mostly a commit of `php artisan ide_helper:models`.

Comment (qwerty287) [reply app/Models/BaseAlbumImpl.php:108]: Can you just remove it or is it still necessary?

Comment (ildyria) [reply app/Models/BaseAlbumImpl.php:108]: I think there was a risk of column removal which would mess up with the sqlite DB. :/

**Review by qwerty287 — APPROVED** [review]

**Pull request merged**